### PR TITLE
Add-on with long name is causing layout issues

### LINKED ISSUE #2415: Add-on with long name is causing layout issues
Steps to reproduce:
1. Submit an add-on with long name i.e. https://addons-dev.allizom.org/en-US/android/addon/lastpass-password-manager/?src=search (without spaces)
2. Access its details page

Expected results:
There are no display or layout issues.

Actual results
Add-on name is not truncated or rendered as expected.

Notes/Issues:
- add-on details page: long name causes horizontal scrolling
- listing pages: icon is moved from its place

Verified on FF53(Android 6.0.1). Issue is reproducing on AMO-dev and stage.
Screenshots for this issue:
untitled

### PR DESCRIPTION
**Pull request by tsl143**
FIX  #2415

### CODE REVIEW

File: src/core/css/SearchResult.scss [hunk 1]
```diff
@@ -34,10 +34,16 @@
   margin: 0;
   padding: 0;
   text-decoration: none;
+  width: calc(100% - 40px);
```
Comment (muffinresearch) [inline src/core/css/SearchResult.scss]: To aid us in the future can you add comments for what the calc is based on in both cases?

**Comment by muffinresearch** [comment]
@tsl143 thanks for the patch. Please can you add before and after screenshots to the pr?

File: src/core/css/SearchResult.scss (hunk 1 above)
Comment (tsl143) [inline src/core/css/SearchResult.scss]: For addon listing 
Calc(100% - 40px) - 38 px is fixed width `.SearchResult-icon` hence (100%-40px) for `.SearchResult-heading`

**Comment by tsl143** [comment]
@muffinresearch Screenshots 

## FOR ADDONS LISTING
**Before**
before

**After**
after


## FOR THEMES LISTING
**Before**
before

**After**
after

File: src/core/css/SearchResult.scss (hunk 1 above)
Comment (muffinresearch) [inline src/core/css/SearchResult.scss]: @tsl143 thanks, could you add those details to the CSS as comments please.

**Comment by muffinresearch** [comment]
Is there a way to do this with flexbox instead of all the floats and calc? 

I appreciate the floats were there before but I think this might be cleaner with flexbox directly.

**Comment by muffinresearch** [comment]
There could do with being more whitespace between the text + ratings, so I'd allow for at least 10px gutter between them so things don't look too cramped.

**Comment by tsl143** [comment]
We can for sure use flexbox (always like them )
Right now I am on vacation for a week, will that be fine if i work on it once I am back?

**Comment by muffinresearch** [comment]
> We can for sure use flexbox (always like them )
Right now I am on vacation for a week, will that be fine if i work on it once I am back?

Yep that should be fine.

**Comment by muffinresearch** [comment]
@tsl143 hi, hope you had a good vacation. Let me know if you have any updates to review.

**Comment by tsl143** [comment]
Thanks, yes indeed I had a really good one :)
I started working on the bug, after pull, when I run `yarn dev:amo` it serves API from `localhost:3000` instead of amo dev server.
I didnt setup addons-server repo, I want to use APIs from dev server as before.

**Comment by muffinresearch** [comment]
@tsl143 you can configure the api host with a local config see https://github.com/mozilla/addons-frontend#configuring-for-local-development for details. 

This example will work for -dev:

```
// config/local-development.js
module.exports = { 
  apiHost: 'https://addons-dev.allizom.org',
  amoCDN: 'https://addons-dev-cdn.allizom.org',
};
```

**Comment by tsl143** [comment]
This is what I was looking for, thanks :)
I will raise PR asap.

**Comment by muffinresearch** [comment]
@tsl143 feel free to just update this PR.

**Comment by tsl143** [comment]
Screenshot after

after_flex

File: src/core/css/SearchResult.scss [hunk 2]
```diff
@@ -96,24 +102,28 @@
...
+
 .SearchResult-icon {
-  display: inline-block;
-  float: right;
   height: 38px;
+  margin-left: 5px;
   width: 38px;
 
+  [dir=rtl] & {
+    margin: 0 5px 0 0;
```
Comment (tofumatt) [inline src/core/css/SearchResult.scss]: We have a mixin for this in `cire/css/inc/mixins`: `@include margin-start(5px);`

Comment (tofumatt) [inline src/core/css/SearchResult.scss]: (It would go in the main selector though, where `margin-left` is.

File: src/core/css/SearchResult.scss [hunk 3]
```diff
@@ -32,12 +32,18 @@
   font-size: $font-size-default;
   font-weight: bold;
   margin: 0;
+  min-width: 1%;
```
Comment (tofumatt) [inline src/core/css/SearchResult.scss]: Why is this needed?

Comment (tsl143) [inline src/core/css/SearchResult.scss]: The flex-item is overflowing so to control that.

Comment (tsl143) [inline src/core/css/SearchResult.scss]: Reference:
https://stackoverflow.com/questions/12022288/how-to-keep-a-flex-item-from-overflowing-due-to-its-text#answer-12131365

**Comment by tsl143** [comment]
ping @tofumatt , @muffinresearch

**Comment by tsl143** [comment]
The issue has already been fixed by layout change, hence closing
